src/gc/none.cr

```diff
@@ -136,4 +137,85 @@ module GC
   # :nodoc:
   def self.push_stack(stack_top, stack_bottom)
   end
+
+  # Stop and start the world.
+  #
+  # This isn't a GC-safe stop-the-world implementation (it may allocate objects
+  # while stopping the world), but the guarantees are enough for the purpose of
+  # gc_none.
+  #
+  # Thread safety is guaranteed by the mutex in Crystal::PointerLinkedList:
+  # either a thread is starting and hasn't added itself to the list (it will
+  # block until it can acquire the lock), or is currently adding itself (the
+  # current thread will block until it can acquire the lock).
+  #
+  # In both cases there can't be a deadlock since we won't suspend another
+  # thread until it has successfuly added (or removed) itself to the linked
+  # list and released the lock, and the other thread won't progress until it
+  # can add (or remove) itself from the list.
+  #
+  # Finally, we lock the mutex and keep it locked until we resume the world,
+  # so any thread waiting on the mutex will only be resumed when the world is
+  # resumed.
+
+  # :nodoc:
+  class_property(sig_suspend : Int32) do
```

Comment:
Why are these properties? Are they expected to change? Can't they be constants?

Reply:
gc/boehm allows them to be configured, so I kinda copied that idea... but we never had the need to, so they should probably be just an internal implementation detail :+1: